swupd/manifest.go

```diff
@@ -755,22 +756,60 @@ func (m *Manifest) addManifestFiles(ui UpdateInfo, c config) error {
 		}
 	} else {
 		// Add files to manifest that do not exist in included bundles.
+		chrootDir := filepath.Join(c.imageBase, fmt.Sprint(ui.version), "full")
 		includes := m.GetRecursiveIncludes()
+		usedDirs := make(map[string]struct{})
+		bundleDirs := []string{}
+		bundleFiles := []string{}
 		for f := range m.BundleInfo.Files {
+			fullPath := filepath.Join(chrootDir, f)
+			if fi, err := os.Lstat(fullPath); err == nil {
+				if !fi.IsDir() {
+					usedDirs[filepath.Dir(f)] = struct{}{}
```

Comment:
Do we need any special handling for a file `f` in the root dir, e.g. `/foo`? I assume Dir would return `./` in that case, but the only non-dirs in root should be the symlinks `lib`, `lib64`, `bin`, and `sbin`, right?

Reply:
`filepath.Dir("/file")` will return `"/"` which is correct in my mind though won't come up as mixer doesn't put `/` in the manifests so it wouldn't be checked for removal.